packages/aws-cdk-lib/aws-ecs/lib/amis.ts

```diff
@@ -267,6 +287,14 @@ export class EcsOptimizedImage implements ec2.IMachineImage {
     } else {
       throw new Error('This error should never be thrown');
     }
+    const kernelPath = (() => {
+      if (!props.kernel) {
+        return '';
+      } else if (props.generation !== ec2.AmazonLinuxGeneration.AMAZON_LINUX_2) {
+        throw new Error('Kernel version can only be specified for Amazon Linux 2');
+      }
+      return props.kernel.toString() + '/';
+    })();
```

Comment:
```suggestion
    if (props.kernel && props.generation !== ec2.AmazonLinuxGeneration.AMAZON_LINUX_2) {
      throw new Error('Kernel version is supported only for Amazon Linux 2');
    }
```
I'd prefer to keep it simple instead of using a function.
We can the concatenate the string with `(props.kernel ? `${props.kernel}/` : '')`.

Reply:
Thanks for the suggestion. I will fix it accordingly. It will look like:
```ts
+ (props.kernel ? props.kernel.toString() + '/' : '')
```

Reply:
Looks good, thannks.